router/router.go

```diff
@@ -101,6 +106,33 @@ func (r *router) GetClient(key string) (client interface{}, isRemote bool, err e
 		return nil, false, err
 	}
 
+	return r.getClientByHost(dest)
+}
+
+// ClientResult is a struct that contains a reference to the actual callable
+// client and a bool indicating whether or not that client is local or remote.
+type ClientResult struct {
+	Client   interface{}
+	IsRemote bool
+}
+
+func (r *router) GetNClients(key string, n int) (clients []ClientResult, err error) {
+	dests, err := r.ringpop.LookupN(key, n)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dest := range dests {
+		client, isRemote, err := r.getClientByHost(dest)
+		if err != nil {
+			return nil, err
+		}
+		clients = append(clients, ClientResult{client, isRemote})
```

Comment:
nit; depending on how often this code is called you might want to consider allocating a slice with the capacity of `len(dests)` before the loop to prevent excessive allocations in the hot path.